src/Orleans/Configuration/ConfigurationExtensions.cs

```diff
@@ -0,0 +1,32 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace Orleans.Runtime.Configuration
+{
+    public static class ConfigurationExtensions
+    {
+        /// <summary>
+        /// Applies the specified config change defined by <paramref name="nodeConfigUpdater"/> to 
+        /// <see cref="ClusterConfiguration.Defaults"/> and all the node configurations currently 
+        /// defined in <see cref="ClusterConfiguration.Overrides"/>.
+        /// </summary>
+        /// <param name="config">The cluster configuration object to add provider to.</param>
+        /// <param name="nodeConfigUpdater">The function to apply to each node configuration.</param>
+        public static void ApplyToAllNodes(this ClusterConfiguration config, Action<NodeConfiguration> nodeConfigUpdater)
+        {
+            foreach (var nodeConfiguration in config.GetDefinedNodeConfigurations())
+            {
+                nodeConfigUpdater.Invoke(nodeConfiguration);
+            }
+        }
+
+        private static IEnumerable<NodeConfiguration> GetDefinedNodeConfigurations(this ClusterConfiguration config)
```

Comment:
Since the property is named `Overrides`, I think it'd be less confusing if we stick to that: `GetOverrides` or `GetOverrideConfigurations` instead of `GetDefinedNodeConfigurations` and `UpdateOverrides or `UpdateOverrideConfiguration` instead of `ApplyToAllNodes`.